library/spdm_requester_lib/libspdm_req_get_measurement_extension_log.c

```diff
@@ -0,0 +1,426 @@
+        LIBSPDM_ASSERT (message_size >= transport_header_size);
+        spdm_response = (void *)(message);
+        spdm_response_size = message_size;
+
+        libspdm_zero_mem(spdm_response, spdm_response_size);
+        status = libspdm_receive_spdm_response(spdm_context, session_id,
+                                               &spdm_response_size,
+                                               (void **)&spdm_response);
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_RECEIVE_FAIL;
+            goto done;
+        }
+
+        /* -=[Validate Response Phase]=- */
+        if (spdm_response_size < sizeof(spdm_message_header_t)) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto done;
+        }
+        if (spdm_response->header.spdm_version != spdm_request->header.spdm_version) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+            goto done;
+        }
+        if (spdm_response->header.request_response_code == SPDM_ERROR) {
+            status = libspdm_handle_error_response_main(
+                spdm_context, session_id,
+                &spdm_response_size,
+                (void **)&spdm_response, SPDM_GET_MEASUREMENT_EXTENSION_LOG,
+                SPDM_MEASUREMENT_EXTENSION_LOG);
+            if (LIBSPDM_STATUS_IS_ERROR(status)) {
+                libspdm_release_receiver_buffer (spdm_context);
+                goto done;
+            }
+        } else if (spdm_response->header.request_response_code != SPDM_MEASUREMENT_EXTENSION_LOG) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+            goto done;
+        }
+        if (spdm_response_size < sizeof(spdm_measurement_extension_log_response_t)) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto done;
+        }
+        if ((spdm_response->portion_length > spdm_request->length) ||
+            (spdm_response->portion_length == 0)) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+            goto done;
+        }
+        if (spdm_response_size < sizeof(spdm_measurement_extension_log_response_t) +
+            spdm_response->portion_length) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto done;
+        }
+        if (spdm_response->portion_length > 0xFFFF - spdm_request->offset) {
+            libspdm_release_receiver_buffer (spdm_context);
+            status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+            goto done;
+        }
+        if (spdm_response->remainder_length > 0xFFFF - spdm_request->offset -
```

Comment:
why use 0xFFFF ?

Reply:
Thanks. I have changed it to 0xFFFFFFFF.

Reply:
Thanks. Please add unit test.